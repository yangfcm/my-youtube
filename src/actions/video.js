import { defAxios as axios } from "../settings";
import { FETCH_VIDEOS, FETCH_VIDEO, CATCH_ERROR } from "./types";
import { DEFAULT_ERROR_MSG } from "./default-error-msg";

/**
 * Get a set of videos based on filter,
 * e.g. videos belonging to a channel or videos belonging to a play list etc.
 */
export const fetchVideos = (filter, pageToken) => {
  return async (dispatch) => {
    try {
      const response = await axios.get("/videos", {
        params: {
          ...axios.defaults.params,
          ...filter,
          part: "snippet,statistics",
          maxResults: 15,
          pageToken,
        },
      });
      dispatch({
        type: FETCH_VIDEOS,
        payload: response.data,
      });
    } catch (e) {
      dispatch({
        type: CATCH_ERROR,
        payload: {
          ...e.response.data.error,
          displayMessage: DEFAULT_ERROR_MSG.FAILED_TO_FETCH_VIDEO,
        },
      });
    }
  };
};

/**
 * Get a video by id
 */
export const fetchVideo = (videoId) => {
  return async (dispatch) => {
    try {
      const response = await axios.get("/videos", {
        params: {
          ...axios.defaults.params,
          part: "snippet,statistics",
          id: videoId,
        },
      });
      dispatch({
        type: FETCH_VIDEO,
        payload: response.data,
      });
    } catch (e) {
      dispatch({
        type: CATCH_ERROR,
        payload: {
          ...e.response.data.error,
          displayMessage: DEFAULT_ERROR_MSG.FAILED_TO_FETCH_VIDEO,
        },
      });
    }
  };
};
