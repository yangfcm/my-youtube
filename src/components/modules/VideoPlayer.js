import React from "react";

const VideoPlayer = ({ videoId }) => {
  const videoSrc = `https://www.youtube.com/embed/${videoId}`;
  const stylePlayerContainer = {
    position: "relative",
    maxWidth: "100%",
    height: "0",
    overflow: "hidden",
    background: "#dcddde",
    paddingBottom: "62%"
  };

  const stylePlayer = {
    width: "100%",
    height: "100%",
    position: "absolute"
  };

  return (
    <div>
      <div style={stylePlayerContainer} className="mb-3">
        <iframe
          src={videoSrc}
          alt="video"
          title="video player"
          style={stylePlayer}
        ></iframe>
      </div>
    </div>
  );
};

export default VideoPlayer;
