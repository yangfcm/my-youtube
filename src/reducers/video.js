import { FETCH_VIDEOS, FETCH_VIDEO } from "../actions/types";

export const videoReducer = (state = {}, action) => {
  switch (action.type) {
    case FETCH_VIDEOS:
      return { ...state, videos: action.payload };
    case FETCH_VIDEO:
      return { ...state, video: action.payload };
    default:
      return state;
  }
};
