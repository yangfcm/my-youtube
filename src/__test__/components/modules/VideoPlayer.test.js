import React from "react";
import { shallow } from "enzyme";
import VideoPlayer from "components/modules/VideoPlayer";
import { videoId } from "__test__/fixtures/video";

describe("Test VideoPlayer component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<VideoPlayer videoId={videoId} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it("should render component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render iframe with correct url", () => {
    const iframe = wrapper.find("iframe");
    expect(iframe.exists()).toBe(true);
    expect(iframe.prop("src")).toBe(`https://www.youtube.com/embed/${videoId}`);
  });
});
