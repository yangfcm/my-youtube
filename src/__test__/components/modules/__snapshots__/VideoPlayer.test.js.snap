// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test VideoPlayer component should render component 1`] = `
<div>
  <div
    className="mb-3"
    style={
      Object {
        "background": "#dcddde",
        "height": "0",
        "maxWidth": "100%",
        "overflow": "hidden",
        "paddingBottom": "62%",
        "position": "relative",
      }
    }
  >
    <iframe
      alt="video"
      src="https://www.youtube.com/embed/hQq7ZSgvhtU"
      style={
        Object {
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        }
      }
      title="video player"
    />
  </div>
</div>
`;
